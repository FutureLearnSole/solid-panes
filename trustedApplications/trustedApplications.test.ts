const $rdf = require('rdflib');
const ns = require('solid-namespace')($rdf);
const { getStatementsToDelete, getStatementsToAdd } = require('./trustedApplicationsUtils');

describe('getStatementsToDelete', () => {
  it('should return an empty array when there are no statements', () => {
    const mockStore = $rdf.graph();
    const mockOrigin = $rdf.lit('Mock origin');
    const mockProfile = $rdf.lit('Mock profile');
    expect(getStatementsToDelete(mockOrigin, mockProfile, mockStore, ns)).toEqual([]);
  });

  it('should return all statements for the given origin', () => {
    const mockStore = $rdf.graph();
    const mockApplication = $rdf.lit('Mock application');
    const mockOrigin = $rdf.lit('Mock origin');
    const mockProfile = $rdf.lit('Mock profile');
    mockStore.add(mockApplication, ns.acl('origin'), mockOrigin);
    mockStore.add(mockApplication, ns.acl('mode'), ns.acl('Read'));
    mockStore.add(mockProfile, ns.acl('trustedApp'), mockApplication);
    const statementsToDelete = getStatementsToDelete(mockOrigin, mockProfile, mockStore, ns);
    expect(statementsToDelete.length).toBe(3);
    expect(statementsToDelete).toMatchSnapshot();
  });

  it('should not return statements for a different origin', () => {
    const mockStore = $rdf.graph();
    const mockApplication = $rdf.lit('Mock application');
    const mockOrigin = $rdf.lit('Mock origin');
    const mockProfile = $rdf.lit('Mock profile');
    mockStore.add(mockApplication, ns.acl('origin'), mockOrigin);
    mockStore.add(mockApplication, ns.acl('mode'), ns.acl('Read'));
    mockStore.add(mockProfile, ns.acl('trustedApp'), mockApplication);

    const statementsToDelete = getStatementsToDelete($rdf.lit('A different origin'), mockProfile, mockStore, ns);
    expect(statementsToDelete.length).toBe(0);
    expect(statementsToDelete).toEqual([]);
  });
});

describe('getStatementsToAdd', () => {
  it('should return all required statements to add the given permissions for a given origin', () => {
    const mockOrigin = $rdf.lit('Mock origin');
    const mockProfile = $rdf.lit('Mock profile');
    const modes = [ns.acl('Read'), ns.acl('Write')];

    const statementsToAdd = getStatementsToAdd(mockOrigin, 'mock_app_id', modes, mockProfile, ns);
    expect(statementsToAdd.length).toBe(4);
    expect(statementsToAdd).toMatchSnapshot();
  });

});
