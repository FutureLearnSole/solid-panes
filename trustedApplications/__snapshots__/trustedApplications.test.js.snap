// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStatementsToAdd should return all required statements to add the given permissions for a given origin 1`] = `
Array [
  Statement {
    "object": BlankNode {
      "id": "bn_mock_app_id",
      "termType": "BlankNode",
      "value": "bn_mock_app_id",
    },
    "predicate": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#trustedApp",
    },
    "subject": Literal {
      "termType": "Literal",
      "value": "Mock profile",
    },
    "why": Literal {
      "termType": "Literal",
      "value": "Mock profile",
    },
  },
  Statement {
    "object": Literal {
      "termType": "Literal",
      "value": "Mock origin",
    },
    "predicate": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#origin",
    },
    "subject": BlankNode {
      "id": "bn_mock_app_id",
      "termType": "BlankNode",
      "value": "bn_mock_app_id",
    },
    "why": Literal {
      "termType": "Literal",
      "value": "Mock profile",
    },
  },
  Statement {
    "object": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#Read",
    },
    "predicate": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#mode",
    },
    "subject": BlankNode {
      "id": "bn_mock_app_id",
      "termType": "BlankNode",
      "value": "bn_mock_app_id",
    },
    "why": Literal {
      "termType": "Literal",
      "value": "Mock profile",
    },
  },
  Statement {
    "object": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#Write",
    },
    "predicate": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#mode",
    },
    "subject": BlankNode {
      "id": "bn_mock_app_id",
      "termType": "BlankNode",
      "value": "bn_mock_app_id",
    },
    "why": Literal {
      "termType": "Literal",
      "value": "Mock profile",
    },
  },
]
`;

exports[`getStatementsToDelete should return all statements for the given origin 1`] = `
Array [
  Statement {
    "object": Literal {
      "termType": "Literal",
      "value": "Mock application",
    },
    "predicate": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#trustedApp",
    },
    "subject": Literal {
      "termType": "Literal",
      "value": "Mock profile",
    },
    "why": NamedNode {
      "termType": "NamedNode",
      "value": "chrome:theSession",
    },
  },
  Statement {
    "object": Literal {
      "termType": "Literal",
      "value": "Mock origin",
    },
    "predicate": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#origin",
    },
    "subject": Literal {
      "termType": "Literal",
      "value": "Mock application",
    },
    "why": NamedNode {
      "termType": "NamedNode",
      "value": "chrome:theSession",
    },
  },
  Statement {
    "object": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#Read",
    },
    "predicate": NamedNode {
      "termType": "NamedNode",
      "value": "http://www.w3.org/ns/auth/acl#mode",
    },
    "subject": Literal {
      "termType": "Literal",
      "value": "Mock application",
    },
    "why": NamedNode {
      "termType": "NamedNode",
      "value": "chrome:theSession",
    },
  },
]
`;
